modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -294,48 +382,142 @@ export default class GeoJsonLayer extends CompositeLayer {
         layerProps.lines
       );
 
+    let pointLayerProps;
+    let pointLayerUpdateTriggers;
+    if (pointType === 'circle') {
+      pointLayerProps = {
+        stroked,
+        filled,
+        radiusUnits: sizeUnits,
+        radiusScale: sizeScale,
+        radiusMinPixels: sizeMinPixels,
+        radiusMaxPixels: sizeMaxPixels,
+        lineWidthUnits,
+        lineWidthScale,
+        lineWidthMinPixels,
+        lineWidthMaxPixels,
+
+        getFillColor: this.getSubLayerAccessor(getFillColor),
+        getLineColor: this.getSubLayerAccessor(getLineColor),
+        getRadius: this.getSubLayerAccessor(getRadius),
+        getLineWidth: this.getSubLayerAccessor(getLineWidth),
+
+        transitions: transitions && {
+          getPosition: transitions.geometry,
+          getFillColor: transitions.getFillColor,
+          getLineColor: transitions.getLineColor,
+          getRadius: transitions.getRadius,
+          getLineWidth: transitions.getLineWidth
+        }
+      };
+      pointLayerUpdateTriggers = {
+        getFillColor: updateTriggers.getFillColor,
+        getLineColor: updateTriggers.getLineColor,
+        getRadius: updateTriggers.getRadius,
+        getLineWidth: updateTriggers.getLineWidth
+      };
+    } else if (pointType === 'icon') {
+      pointLayerProps = {
+        alphaCutoff,
+        billboard,
+        iconAtlas,
+        iconMapping,
+        onIconError,
+        sizeMaxPixels,
+        sizeMinPixels,
+        sizeScale,
+        sizeUnits,
+
+        getAngle: this.getSubLayerAccessor(getAngle),
+        getColor: this.getSubLayerAccessor(getColor),
```

Comment:
If circle uses `getFillColor` then icon and text should too

Reply:
Done